tests/test_http_request.py

```diff
@@ -1147,5 +1149,65 @@ def test_latin1(self):
         self._test_request(params=(u'pas£',), encoding='latin1')
 
 
+class JSONRequestTest(RequestTest):
+    request_class = JSONRequest
+    default_method = 'GET'
+    default_headers = {b'Content-Type': [b'application/json'], b'Accept': [b'application/json, text/javascript, */*; q=0.01']}
+
+    def setUp(self):
+        warnings.simplefilter("always")
+        super(JSONRequestTest, self).setUp()
+
+    def test_data(self):
```

Comment:
would you mind split this test method into several methods, for individual cases?

Reply:
Also, could you please add a test for .replace method?

Reply:
Test for replacing data file done. Testing for replacing body is already in the base RequestTest class which JSONRequestTest class inherits from.